### ISSUE TITLE
Disable MD031 (space around code blocks) for lists?

### ISSUE THREAD

**Issue by borekb** [DavidAnson/markdownlint#200]
The "MD031 – Fenced code blocks should be surrounded by blank lines" rule is invoked also for code blocks that are inside lists. (It was even reported as a bug when it wasn't: #189.) However, I'm not sure that it's desired because adding a blank line actually influences how the _other_ `<li>` elements are treated (with a blank line, _all_ of them are wrapped in an additional `<p>` tag).

Compare:

## Example without a blank line

**Source**:

```md
- List
- with
- code block:
    ```js
    console.log('hello world');
    ```
```

**Output**:

- List
- with
- code block:
    ```js
    console.log('hello world');
    ```

## Example with a blank line

**Source**:

```md
- List
- with
- code block:

    ```js
    console.log('hello world');
    ```
```

**Output**:

- List
- with
- code block:

    ```js
    console.log('hello world');
    ```

---

I'd appreciate if there was a way to disable MD031 for lists.

**Comment by DavidAnson** [comment]
Good idea!

**Comment by borekb** [comment]
Thanks for implementing this! For anyone reading this, since 0.16.0, this config resolves it:

```json
{
  "MD031": { "list_items": false }
}
```

See https://github.com/DavidAnson/markdownlint/commit/c5b74d22b932c6be5605219af0069255f0adaa39.
